inc/dg/functors.h

```diff
@@ -945,135 +1050,168 @@ struct Vortex
         double tpi=2.*M_PI, tpi2=tpi*tpi;
-        double k0= tpi*eddysize_/RZm;
-        double Rmh = Rm_/2.;
-        double Zmh = Zm_/2.;
+        double k0= tpi*L_E_/N_k;
+        double N_kRh = N_kR_/2.;
+        double N_kZh = N_kZ_/2.;
 
         thrust::random::minstd_rand generator;
         thrust::random::normal_distribution<double> ndistribution;
         thrust::random::uniform_real_distribution<double> udistribution(0.0,tpi);
-        for (unsigned j=1;j<=Zm_;j++)
+        for (unsigned j=1;j<=N_kZ_;j++)
         {
-            double kZ2=tpi2*(j-Zmh)*(j-Zmh)/(Zm2);
-            for (unsigned i=1;i<=Rm_;i++)
+            double kZ2=tpi2*(j-N_kZh)*(j-N_kZh)/(N_kZ2);
+            for (unsigned i=1;i<=N_kR_;i++)
             {
-                double kR2=tpi2*(i-Rmh)*(i-Rmh)/(Rm2);
-                int z=(j-1)*(Rm_)+(i-1);
+                double kR2=tpi2*(i-N_kRh)*(i-N_kRh)/(N_kR2);
+                int z=(j-1)*(N_kR_)+(i-1);
                 kvec[z]= sqrt(kR2 + kZ2);  //radial k number
                 sqEkvec[z]=pow(kvec[z]*4.*k0/(kvec[z]+k0)/(kvec[z]+k0),gamma_/2.); //Energie in k space with max at 1.
                 unif1[z]=cos(udistribution(generator));
                 unif2[z]=sin(udistribution(generator));
                 normal1[z]=ndistribution(generator);
                 normal2[z]=ndistribution(generator);
-                normalamp[z]=sqrt(normal1[z]*normal1[z]+normal2[z]*normal2[z]);
-                normalphase[z]=atan2(normal2[z],normal1[z]);
+                alpha[z]=sqrt(normal1[z]*normal1[z]+normal2[z]*normal2[z]);
+                theta[z]=atan2(normal2[z],normal1[z]);
             }
         }
 
     }
       /**
-     * @brief Return the value of the Bath
+     * @brief Return the value of the Bath 
      *
+       \f[f(R,Z) = A B \sum_\vec{k} \sqrt{E_k} \alpha_k \cos{\left(k \kappa_k + \theta_k \right)} 
+       \f]
+     * with \f[ \mathcal{N} := \sqrt{\frac{2}{N_{k_R} N_{k_Z}}} \\ 
+                k:=\sqrt{k_R^2 + k_Z^2} \\ 
+                k_R:=2 \pi \left( i -N_{k_R}/2\right)/N_{k_R} \\
+                k_Z:=2 \pi \left( j -N_{k_Z}/2\right)/N_{k_Z} \\
+                k_0:=2 \pi L_E / N_k\\ 
+                N_k := \sqrt{N_{k_R}^2 + N_{k_Z}^2} \\
+                E_k:=\left(4 k k_0/(k+k_0)^2\right)^{\gamma} \\ 
+                \alpha_k := \sqrt{\mathcal{N}_1^2 + \mathcal{N}_2^2} \\
+                \theta_k := \arctan{\left(\mathcal{N}_2/\mathcal{N}_1\right)} \\
+                \kappa_k(R,Z) := (R-R_{min}) \mathcal{U}_1 + (Z-Z_{min}) \mathcal{U}_2  \\
+                \f]
+     * where \f$\mathcal{N}_{1,2}\f$ are random normal distributed real numbers with a mean of \f$\mu = 0\f$ and a standard deviation of \f$\sigma=1 \f$,
+     * \f$\mathcal{U}_{1,2}\f$ are random uniformly distributed real numbers  \f$\in \left[0, 2 \pi \right) \f$ and \f$ A \f$ is the amplitude
      * @param R R - coordinate
      * @param Z Z - coordinate
      *
+     * @return the above function value
      */
+    DG_DEVICE
```

Comment:
da darf kein DG_DEVICE stehen weils  auf der GPU nicht ausführt auf Grund der std::vector in den private variables